src/observations/process_radar.jl

```diff
@@ -899,81 +695,82 @@ function delay(station_code::Int, t_r_utc::DateTime,
+        τe = compute_delay(observatory, t_r_utc,  tc/2, niter, eo=eo, xve=xve, xvs=xvs, xva=xva)
+        τn = compute_delay(observatory, t_r_utc,   0.0, niter, eo=eo, xve=xve, xvs=xvs, xva=xva)
+        τs = compute_delay(observatory, t_r_utc, -tc/2, niter, eo=eo, xve=xve, xvs=xvs, xva=xva)
         # Time delay, Doppler shift
         return τn, -F_tx*((τe-τs)/tc)
     end
 
 end
 
-@doc raw"""
-    delay_doppler(astradarfile::String, niter::Int=10; eo::Bool=true, tc::Real=1.0,
-                  xve=earth_pv, xvs=sun_pv, xva=apophis_pv_197, autodiff::Bool=true,
-                  tord::Int=10)
-
-Returns a `DataFrame` with the time delays and Doppler shifts associated with the radar
-measurements of file `astradarfile`.
-
-See also [`process_radar_data_jpl`](@ref).
-
-# Arguments
+function radar_astrometry(radar::RadarJPL{T}, niter::Int = 10; eo::Bool = true, tc::Real = 1.0, xve = earth_pv, 
+                       xvs = sun_pv, xva = apophis_pv_197, autodiff::Bool = true, tord::Int = 10) where {T <: AbstractFloat}
+    return radar_astrometry(radar.rcvr, radar.date, radar.freq, niter; eo = eo, tc = tc, xve = xve, xvs = xvs, xva = xva, autodiff = autodiff, tord = tord)
+end 
 
-- `astradarfile::String`: radar data file.
-- `niter::Int`: number of light-time solution iterations.
-- `eo`: wheter to use `EarthOrientation` or not.
-- `tc`: time offset wrt echo reception time, to compute Doppler shifts by range differences (seconds).
-- `xve`: Earth ephemeris wich takes TDB seconds since J2000 as input and returns Earth barycentric position in km and velocity in km/second.
-- `xvs`: Sun ephemeris wich takes TDB seconds since J2000 as input and returns Sun barycentric position in km and velocity in km/second.
-- `xva`: asteroid ephemeris wich takes TDB seconds since J2000 as input and returns asteroid barycentric position in km and velocity in km/second.
-- `autodiff`: wheter to use the automatic differentiation method of [`delay`](@ref) or not.
-- `tord`: order of Taylor expansions.
-"""
-function delay_doppler(astradarfile::String,
-        niter::Int=10; eo::Bool=true, tc::Real=1.0, xve=earth_pv, xvs=sun_pv,
-        xva=apophis_pv_197, autodiff::Bool=true, tord::Int=10)
+function radar_astrometry(astradarfile::String, niter::Int=10; eo::Bool=true, tc::Real=1.0, xve=earth_pv, xvs=sun_pv,
+                       xva=apophis_pv_197, autodiff::Bool=true, tord::Int=10)
 
+    # Check that astradarfile is a file 
+    @assert isfile(astradarfile) "Cannot open file: $astradarfile"
     # Read radar measurements
-    astradardata = process_radar_data_jpl(astradarfile)
-    #
-    et1 = str2et(string(astradardata[1].utcepoch))
-    #
+    astradardata = read_radar_jpl(astradarfile)
+
+    # UTC time of first radar observation
+    utc1 = astradardata[1].date
+    # TDB seconds since J2000.0 for first astrometric observation
+    et1 = datetime2et(utc1)
+    # Asteroid ephemeris at et1
     a1_et1 = xva(et1)[1]
-    #
+    # Tipe of asteroid ephemeris 
```

Comment:
"Type"

Reply:
Sorry for the typo, it was corrected in the last commit.